### PR TITLE
Plugin review; disable unnecessary plugins

### PR DESCRIPTION
**Pull request by rnelson0**
This is a breaking change, scheduled for v3.0.0.

### CODE REVIEW

File: lib/puppet-lint/plugins/check_classes.rb [hunk 1]
```diff
@@ -71,6 +71,7 @@ def check
     end
   end
 end
+PuppetLint.configuration.send('names_containing_dash')
```
Comment (binford2k) [inline lib/puppet-lint/plugins/check_classes.rb]: did you mean `'disable_names_containing_dash'`

**Comment by binford2k** [comment]
Could you attach a quick list of the rationale for each plugin being disabled?

**Comment by rnelson0** [comment]
Rationale from #561 
All plugins are on by default. Over time, the lint plugins have drifted from the style guide. We reviewed all the core checks and correlated them to the style guide. The checks below either cannot be correlated, require a style guide addition/modification, or are not part of linting. I have commented them for disabling, removing, or requiring a style guide update.

check_classes.rb
- _[names_containing_dash](http://puppet-lint.com/checks/names_containing_dash)_ **Remove**, handled by `puppet parser validate`
- _[code_on_top_scope](http://puppet-lint.com/checks/code_on_top_scope)_ **Style Guide Update**, no matching style guide reference but known good practice

check_documentation.rb
- _[documentation](http://puppet-lint.com/checks/documentation)_ **Style Guide Update**, https://docs.puppet.com/guides/style_guide.html#public-and-private needs improvement in the recommendation

check_nodes.rb
- _[unquoted_node_name](http://puppet-lint.com/checks/unquoted_file_mode)_ **Style Guide Update**, no matching style guide reference but known good practice

check_resources.rb
- _[duplicate_params](http://puppet-lint.com/checks/duplicate_params)_ **Remove**, handled by `puppet parser validate`

check_strings.rb
- _[only_variable_string](http://puppet-lint.com/checks/only_variable_string)_ **Disable**, style guide reference in discussion in [DOCUMENT-274](https://tickets.puppetlabs.com/browse/DOCUMENT-274)
- _[quoted_booleans](http://puppet-lint.com/checks/puppet_url_without_modules/)_ **Style Guide Update**, no matching style guide reference but known good practice
- _[puppet_url_without_modules](http://puppet-lint.com/checks/puppet_url_without_modules/)_ **Disable**, no style guide reference

check_variables.rb
- _[variable_contains_dash](http://puppet-lint.com/checks/variable_contains_dash)_ **Remove**, handled by `puppet parser validate`
- _[variable_is_lowercase](http://puppet-lint.com/checks/variable_is_lowercase)_ **Remove**, handled by `puppet parser validate`

File: lib/puppet-lint/plugins/check_classes.rb (hunk 1 above)
Comment (rnelson0) [inline lib/puppet-lint/plugins/check_classes.rb]: Fixing

**Pull request merged**